### PR TITLE
Split graphics in 2 libraries a core and a full one

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

- Split graphics in 2 libraries a core and a full one, code cleaning, adding all intellisense comments.

## Motivation and Context

- See discussion nanoframework/Home#1231
- Have a core graphic nuget to reuse in the IoT Device bindings
  - Color
  - Point
  - IFont
  - Display orientation
  - potentially few other ones
- While making some changes, adjusting some other elements of the Graphic libs like comments, creating few enums rather than raw constants (all work in progress in this PR)
- PR on native side done as well with adjustment of version and native declarations after code cleaning.

Second step will be to adjust the Sample repo.
Third step will be to adjust all the IoT Device nugets to include this class. The "old" System.Drawing containing only the Color today will be able to get retired.

## How Has This Been Tested?

This has been tested on a real device with the primitive sample. An ESP32 Wrover.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [x] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by alberk8** [comment]
Should bitmap be in the core?

**Comment by Ellerbach** [comment]
> Should bitmap be in the core?

I cannot because it is deeply linked to the native part. Creating an abrasct or equivalent has not that much sense neither.

File: nanoFramework.Graphics.Core/nanoFramework.Graphics.Core.nfproj [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectTypeGuids>{11A8DD76-328B-46DF-9F39-F559912D0360};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <ProjectGuid>67cd2b6c-e71a-4a98-9c7c-b88109abe158</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <FileAlignment>512</FileAlignment>
+    <RootNamespace>nanoFramework.Graphics.Core</RootNamespace>
+    <AssemblyName>nanoFramework.Graphics.Core</AssemblyName>
+    <TargetFrameworkVersion>v1.0</TargetFrameworkVersion>
+    <DocumentationFile>bin\$(Configuration)\nanoFramework.Graphics.xml</DocumentationFile>
```
Comment (josesimoes) [inline nanoFramework.Graphics.Core/nanoFramework.Graphics.Core.nfproj:18]:
```suggestion
    <DocumentationFile>bin\$(Configuration)\nanoFramework.Graphics.Core.xml</DocumentationFile>
```

File: nanoFramework.Graphics/Input/ButtonEnum.cs [hunk 2]
```diff
@@ -7,371 +7,819 @@
...
+    /// The button or key.
     /// </summary>
     public enum Button
     {
-        /// <summary>  </summary>
+        /// <summary>None.</summary>
         None = 0,
-        /// <summary>  </summary>
+
+        /* Those are the mouse keys 0x01 -> 0x06 */
```
Comment (josesimoes) [inline nanoFramework.Graphics/Input/ButtonEnum.cs:17]: Do we really need mouse keys in nano?? 🤔

File: nanoFramework.Graphics/Input/RawTouchActions.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace nanoFramework.UI.Input
+{
+    /// <summary>
+    /// Raw touch actions.
+    /// </summary>
+    public enum RawTouchActions
+    {
+        /// <summary>
+        ///  Touch Down
```
Comment (josesimoes) [inline nanoFramework.Graphics/Input/RawTouchActions.cs:15]: Missing dots ending comment sentences.

**Review by josesimoes — COMMENTED** [review]
WOW!! This was massive!
A TON of work reworkin, organizing and adding comments (almost!) everywhere.
Impressive work (I'm sure several parts where rather tedious). Thanks for taking care of this.

Just a couple of minor things I've bumped into. 

For a future discussion: a general clean-up is due for things like mouse events and keys, keyboard with function keys, numeric keypad, sessions, etc. All this is bagagge from the NETMF days intended for usage scenarious that we are not (nor planning) to address.

File: nanoFramework.Graphics/Input/ButtonEnum.cs (hunk 2 above)
Comment (Ellerbach) [reply nanoFramework.Graphics/Input/ButtonEnum.cs:17]: yes, we should keep that. Those are the standard key used everywhere. For HID implementation for example, those are the ones you can map things.

Comment (Ellerbach) [reply nanoFramework.Graphics/Input/ButtonEnum.cs:17]: > For a future discussion: a general clean-up is due for things like mouse events and keys, keyboard with function keys, numeric keypad, sessions, etc. All this is bagagge from the NETMF days intended for usage scenarious that we are not (nor planning) to address.

And for those ones, we definitely should KEEP them. You can make a mouse out of nano if you want with a screen attached, why not :-) Again, the codes are HID ones, they are shared across any platform.

**Review by josesimoes — APPROVED** [review]
All good now! Thanks.

**Pull request merged**